enterprise/backend/src/metabase_enterprise/serialization/api.clj

```diff
@@ -98,6 +99,16 @@
                       (when (.exists dst)
                         (io/delete-file dst)))}))
 
+(defn- find-serialization-dir
+  "Find an actual top-level dir with serialization data inside, instead of picking up various .DS_Store and similar
+  things."
+  ^File [^File parent]
+  (->> (.listFiles parent)
+       (u/seek (fn [^File f]
+                 (and (.isDirectory f)
+                      (seq (set/intersection v2.ingest/legal-top-level-paths
+                                             (set (.list f)))))))))
```

Comment:
nitpick: It's a bit shorter and more efficient to check `(some legal-top-level-paths (.list f))`